src/studio/src/designer/backend/Controllers/RepositoryController.cs

```diff
@@ -369,14 +373,24 @@ public async Task<RepositoryModel> CreateApp(string org, string repository)
         public async Task<ActionResult<RepositoryModel>> CopyApp(string org, string sourceRepository, string targetRepository)
         {
             string developer = AuthenticationHelper.GetDeveloperUserName(HttpContext);
-            RepositoryModel repo = await _repository.CopyRepository(org, sourceRepository, targetRepository, developer);
 
-            if (repo.RepositoryCreatedStatus == HttpStatusCode.Created)
+            try
             {
-                return Created(repo.CloneUrl, repo);
-            }
+                RepositoryModel repo = await _repository.CopyRepository(org, sourceRepository, targetRepository, developer);
 
-            return StatusCode((int)repo.RepositoryCreatedStatus);
+                if (repo.RepositoryCreatedStatus == HttpStatusCode.Created)
+                {
+                    return Created(repo.CloneUrl, repo);
+                }
+
+                await _repository.DeleteRepository(org, targetRepository);
```

Comment:
Why deleting the local target repository?

Reply:
This deletes both the remote and local repo, in the case that something goes wrong. This way the user can try again later without there being any resources causing conflict.

Reply:
Sorry, my bad - I missed the return statement within the if.